File: 189-RotateArray/189-RotateArray.py
# Last updated: 09/10/2025, 09:40:08
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        if k != 0 and len(nums)> 1:
            if k > len(nums):
                # for i in range (k):
                #     for j in range (len(nums)-1,-1,-1):
                #         if j == len(nums)-1:
                #             temp=nums[j]
                #         else:
                #             nums[j+1] = nums[j]
                #     nums[0]=temp
                k=k%len(nums)
            
            nums1 = nums[-k:]
            nums2 = nums[:len(nums)-k]
            nums1.extend(nums2)
            nums[:]= nums1
            print (nums)
        

File: 189-RotateArray/189-RotateArray.cpp
// Last updated: 03/03/2026, 20:06:43
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        int n = nums.size();
        if (n == 0) return;
        
        k = k % n; // Handle cases where k > n
        if (k == 0) return;

        // Step 1: Reverse the whole vector
        reverse(nums.begin(), nums.end());
        
        // Step 2: Reverse the first k elements
        reverse(nums.begin(), nums.begin() + k);
        
        // Step 3: Reverse the rest
        reverse(nums.begin() + k, nums.end());
    }
};